File: 0455-assign-cookies/0455-assign-cookies.go
func findContentChildren(g []int, s []int) int {

    slices.Sort(g)
    slices.Sort(s)
    result := 0 

    i := len(s) - 1

    for i >= 0 {
        if s[i] >= g[len(g) - 1] {
            result++
            i--
        }
        g = g[:len(g) - 1]
        if len(g) == 0 {
            break
        }
    }

    return result
}

File: 0455-assign-cookies/0455-assign-cookies.cpp
static const bool Booster = [](){
    std::ios_base::sync_with_stdio(false);
    std::cout.tie(nullptr);
    std::cin.tie(nullptr);
    return true;
}();

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int result = 0;
        while (!s.empty() && !g.empty()) {
            if (s.back() >= g.back()) {
                ++result;
                s.pop_back();
            }
            g.pop_back();
        }
        return result;
    }
};